crosslink/internal/common.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crosslink
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	tools "go.opentelemetry.io/build-tools"
+	"golang.org/x/mod/modfile"
+)
+
+func writeModule(module moduleInfo) error {
+	mfParsed, err := modfile.Parse("go.mod", module.moduleContents, nil)
```

Comment:
This is a great example of why we should use `modfile.File` types directly instead of recreating their functionality. This parse a file without the correct filename so it can use only part of the returned type.